packages/client/src/__tests__/buildNFTAnnotations.test.ts

```diff
@@ -0,0 +1,123 @@
+
+      path.join(__dirname, "libquery_engine-TEST_PLATFORM.LIBRARY_TYPE.node");
+      path.join(process.cwd(), "out/libquery_engine-TEST_PLATFORM.so.node")
+      path.join(__dirname, "schema.prisma");
+      path.join(process.cwd(), "out/schema.prisma")
+    `)
+  })
+
+  it('generates annotations for a schema and multiple engines', () => {
+    const annotations = buildNFTAnnotations(
+      ClientEngineType.Library,
+      ['debian-openssl-1.1.x', 'darwin', 'windows'],
+      'out',
+    )
+
+    expect(normalizePaths(annotations)).toMatchInlineSnapshot(`
+
+      path.join(__dirname, "libquery_engine-TEST_PLATFORM.LIBRARY_TYPE.node");
+      path.join(process.cwd(), "out/libquery_engine-TEST_PLATFORM.so.node")
+
+      path.join(__dirname, "libquery_engine-TEST_PLATFORM.dylib.node");
+      path.join(process.cwd(), "out/libquery_engine-TEST_PLATFORM.dylib.node")
+
+      path.join(__dirname, "query_engine-TEST_PLATFORM.dll.node");
+      path.join(process.cwd(), "out/query_engine-TEST_PLATFORM.dll.node")
+      path.join(__dirname, "schema.prisma");
+      path.join(process.cwd(), "out/schema.prisma")
+    `)
+  })
+})
+
+describe('binary', () => {
+  it('generates annotations for a schema and a single engine', () => {
+    const annotations = buildNFTAnnotations(ClientEngineType.Binary, ['debian-openssl-1.1.x'], 'out')
+
+    expect(normalizePaths(annotations)).toMatchInlineSnapshot(`
+
+      path.join(__dirname, "query-engine-TEST_PLATFORM");
+      path.join(process.cwd(), "out/query-engine-TEST_PLATFORM")
+      path.join(__dirname, "schema.prisma");
+      path.join(process.cwd(), "out/schema.prisma")
+    `)
+  })
+
+  it('generates annotations for a schema and multiple engines', () => {
+    const annotations = buildNFTAnnotations(
+      ClientEngineType.Binary,
+      ['debian-openssl-1.1.x', 'darwin', 'windows'],
+      'out',
+    )
+
+    expect(normalizePaths(annotations)).toMatchInlineSnapshot(`
+
+      path.join(__dirname, "query-engine-TEST_PLATFORM");
+      path.join(process.cwd(), "out/query-engine-TEST_PLATFORM")
+
+      path.join(__dirname, "query-engine-TEST_PLATFORM");
+      path.join(process.cwd(), "out/query-engine-TEST_PLATFORM")
+
+      path.join(__dirname, "query-engine-TEST_PLATFORM");
+      path.join(process.cwd(), "out/query-engine-TEST_PLATFORM")
+      path.join(__dirname, "schema.prisma");
+      path.join(process.cwd(), "out/schema.prisma")
+    `)
+  })
+})
+
+describe('dataproxy', () => {
+  it('generates no annotations', () => {
+    const annotations = buildNFTAnnotations(
+      ClientEngineType.DataProxy,
+      ['debian-openssl-1.1.x', 'darwin', 'windows'],
+      'out',
+    )
+
+    // TODO: when using .toMatchInlineSnapshot(), this fails after updating snapshots.
+    // Probably an issue with the snapshot serializer?
+    expect(normalizePaths(annotations)).toBe(`
+
+`)
+  })
+})
+
+describe('special cases', () => {
+  // TODO: document the reason behind this logic.
```

Comment:
From what I remember, the build image is different from the runtime image, so a different engine is needed.
Not sure what's the current state as of today.

Reply:
Yep, build process was local (debian) but deployment was Lamba (rhel) which lead to people having to do weird things - or us just reacting to `NETLIFY=true` and replacing/adding/removing things to make sure.